AddressValueError: Expected 4 octets in 'None'

https://sentry.io/canonical-pj/conjure-up/issues/336412086/

```
AddressValueError: Expected 4 octets in 'None'
  File "conjureup/controllers/lxdsetup/gui.py", line 28, in set_lxd_info
    iface = ipaddress.IPv4Interface(phys_iface_addr)
  File "python3.6/ipaddress.py", line 1384, in __init__
    IPv4Address.__init__(self, addr[0])
  File "python3.6/ipaddress.py", line 1284, in __init__
    self._ip = self._ip_int_from_string(addr_str)
  File "python3.6/ipaddress.py", line 1118, in _ip_int_from_string
    raise AddressValueError("Expected 4 octets in %r" % ip_str)

AddressValueError: Expected 4 octets in 'None'
```

Getting the same error here. At first, I used `none` when running `lxd init`, but tried reconfiguring LXD's network with [`sudo dpkg-reconfigure -p medium lxd`](https://insights.ubuntu.com/2016/04/07/lxd-networking-lxdbr0-explained/) after encountering this error. It did not resolve it (lxd installed as a snap).

Here is my log output:

```
2017-09-23 09:48:40,500 [DEBUG] conjure-up/canonical-kubernetes - provider.py:371 - LXD query cmd: lxc query --wait -X GET /1.0/storage-pools/default
2017-09-23 09:48:47,726 [DEBUG] conjure-up/canonical-kubernetes - utils.py:599 - Parsing 4: lxdbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
    link/ether b2:9a:21:f0:9a:34 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::b09a:21ff:fef0:9a34/64 scope link 
       valid_lft forever preferred_lft forever
 for IPv4 address
2017-09-23 09:48:47,730 [DEBUG] conjure-up/canonical-kubernetes - events.py:53 - Setting Error at conjureup/events.py:147
2017-09-23 09:48:47,732 [ERROR] conjure-up/canonical-kubernetes - events.py:159 - Unhandled exception in <Task finished coro=<LXDSetupController.set_lxd_info() done, defined at /snap/conjure-up/745/lib/python3.6/site-packages/conjureup/controllers/lxdsetup/gui.py:24> exception=AddressValueError("Expected 4 octets in 'None'",)>
Traceback (most recent call last):
  File "/snap/conjure-up/745/lib/python3.6/site-packages/conjureup/controllers/lxdsetup/gui.py", line 28, in set_lxd_info
    iface = ipaddress.IPv4Interface(phys_iface_addr)
  File "/snap/conjure-up/745/usr/lib/python3.6/ipaddress.py", line 1384, in __init__
    IPv4Address.__init__(self, addr[0])
  File "/snap/conjure-up/745/usr/lib/python3.6/ipaddress.py", line 1284, in __init__
    self._ip = self._ip_int_from_string(addr_str)
  File "/snap/conjure-up/745/usr/lib/python3.6/ipaddress.py", line 1118, in _ip_int_from_string
    raise AddressValueError("Expected 4 octets in %r" % ip_str)
ipaddress.AddressValueError: Expected 4 octets in 'None'
2017-09-23 09:48:47,747 [DEBUG] conjure-up/canonical-kubernetes - __init__.py:21 - Showing dialog for exception: Expected 4 octets in 'None'
2017-09-23 09:48:47,841 [ERROR] conjure-up/canonical-kubernetes - sentry.errors.serializer: manager.py:78 - _ip
Traceback (most recent call last):
  File "/snap/conjure-up/745/lib/python3.6/site-packages/raven/utils/serializer/manager.py", line 76, in transform
    return repr(value)
  File "/snap/conjure-up/745/usr/lib/python3.6/ipaddress.py", line 583, in __repr__
    return '%s(%r)' % (self.__class__.__name__, str(self))
  File "/snap/conjure-up/745/usr/lib/python3.6/ipaddress.py", line 1393, in __str__
    return '%s/%d' % (self._string_from_ip_int(self._ip),
AttributeError: _ip
2017-09-23 09:48:47,844 [ERROR] conjure-up/canonical-kubernetes - sentry.errors.serializer: manager.py:78 - _ip
Traceback (most recent call last):
  File "/snap/conjure-up/745/lib/python3.6/site-packages/raven/utils/serializer/manager.py", line 76, in transform
    return repr(value)
  File "/snap/conjure-up/745/usr/lib/python3.6/ipaddress.py", line 583, in __repr__
    return '%s(%r)' % (self.__class__.__name__, str(self))
  File "/snap/conjure-up/745/usr/lib/python3.6/ipaddress.py", line 1393, in __str__
    return '%s/%d' % (self._string_from_ip_int(self._ip),
AttributeError: _ip
```

--- a/conjureup/controllers/lxdsetup/gui.py
+++ b/conjureup/controllers/lxdsetup/gui.py
@@ -25,7 +25,13 @@ class LXDSetupController(common.BaseLXDSetupController):
         self.set_state('lxd-network-name', network['name'])
         phys_iface_addr = utils.get_physical_network_ipaddr(
             network['name'])
-        iface = ipaddress.IPv4Interface("{}/24".format(phys_iface_addr))
+        try:
+            iface = ipaddress.IPv4Interface("{}/24".format(phys_iface_addr))
+        except ipaddress.AddressValueError:
+            raise LXDSetupControllerError(
+                "Unable to determine ip address of {}, please double check "
+                "`/snap/bin/lxc network edit {}` and make sure an address "
+                "is associated with that bridge.".format(network['name']))
         self.set_state('lxd-network', iface.network)
         self.set_state('lxd-gateway', iface.ip)
         self.set_state('lxd-network-dhcp-range-start',

--- a/conjureup/controllers/lxdsetup/gui.py
+++ b/conjureup/controllers/lxdsetup/gui.py
@@ -29,9 +29,10 @@ class LXDSetupController(common.BaseLXDSetupController):
             iface = ipaddress.IPv4Interface("{}/24".format(phys_iface_addr))
         except ipaddress.AddressValueError:
             raise LXDSetupControllerError(
-                "Unable to determine ip address of {}, please double check "
-                "`/snap/bin/lxc network edit {}` and make sure an address "
-                "is associated with that bridge.".format(network['name']))
+                "Unable to determine ip address of {network}, please double "
+                "check `/snap/bin/lxc network edit {network}` and make "
+                "sure an address is associated with that bridge.".format(
+                    network=network['name']))
         self.set_state('lxd-network', iface.network)
         self.set_state('lxd-gateway', iface.ip)
         self.set_state('lxd-network-dhcp-range-start',

--- a/tox.ini
+++ b/tox.ini
@@ -21,7 +21,7 @@ commands =
     pylint conjureup test tools
 
 [testenv:flake]
-commands = flake8 --ignore E501 {posargs} conjureup test tools
+commands = flake8 --ignore E501,E722,E741 {posargs} conjureup test tools
 deps = flake8
 
 [testenv:docs]

